src/clr-angular/data/datagrid/providers/columns.service.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Subscription } from 'rxjs';
+import { DatagridColumnState } from '../interfaces/column-state.interface';
+import { DatagridRenderStep } from '../enums/render-step.enum';
+import { DatagridRenderOrganizer } from '../render/render-organizer';
+
+@Injectable()
+export class ColumnsService implements OnDestroy {
+  subscriptions: Subscription[] = [];
+  columns: BehaviorSubject<DatagridColumnState>[] = [];
+
+  constructor(private organizer: DatagridRenderOrganizer) {
+    this.subscriptions.push(this.organizer.filterRenderSteps(DatagridRenderStep.RESET).subscribe(() => this.reset()));
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.forEach(sub => sub.unsubscribe());
+  }
+
+  private reset() {
+    this.columns.forEach((column, index) => {
+      this.emitStateChange(index, { width: 0 });
+    });
+  }
+
+  // Helper method to emit a change to a column only when there is an actual diff to process for that column
+  emitStateChange(columnIndex: number, diff: Partial<DatagridColumnState>) {
+    if (!this.columns[columnIndex]) {
+      this.columns[columnIndex] = new BehaviorSubject<DatagridColumnState>({});
+    }
+    const current = this.columns[columnIndex].value;
+    const hasChange = Object.keys(diff).find(key => diff[key] !== current[key]);
```

Comment:
I don't think `.find()` works in IE11. I've used `filter()` instead.

Reply:
Fixed